cocos/core/memop/pool.ts

```diff
@@ -107,15 +115,25 @@ export class Pool<T> extends ScalableContainer {
      * @zh 尝试缩容对象池，以释放内存。
      */
     public tryShrink (): void {
-        if (this._nextAvail >> 1 > this._elementsPerBatch) {
-            if (this._dtor) {
-                for (let i = this._nextAvail >> 1; i <= this._nextAvail; i++) {
-                    this._dtor(this._freePool[i]);
-                }
+        const freeObjectNumber = this._freePool.length;
```

Comment:
Should be `const freeObjectNumber = _nextAvail;`?

Reply:
Yep, should be this._nextAvail + 1.
Fixed.